File: 0204-count-primes/0204-count-primes.java
class Solution {
    public int countPrimes(int n) {
        boolean[] isPrime = new boolean[n+1];
        int count = 0;
        for (int i=2; i<n; i++) {
            isPrime[i] = !isPrime[i];
            if (isPrime[i]) {
                count++;
                for (int j=i*2; j<n; j+=i) {
                    isPrime[j] = true;
                }
            }
        }
        return count;
    }
}

File: 0204-count-primes/0204-count-primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        vector<bool> isPrime(n, true);

        if (n <= 2)
            return 0;

        isPrime[0] = isPrime[1] = false;

        for (int i = 2; 1LL * i * i < n; i++) {
            if (isPrime[i]) {
                for (long long j = 1LL * i * i; j < n; j += i) {
                    isPrime[j] = false;
                }
            }
        }

        int count = 0;

        for (int i = 2; i < n; i++) {
            if (isPrime[i])
                count++;
        }

        return count;
    }
};